tools/SourceKit/tools/sourcekitd/include/sourcekitd/CompactArray.h

```diff
@@ -102,7 +102,9 @@ class CompactArrayReaderImpl {
   CompactArrayReaderImpl(void *Buf) : Buf(Buf) {}
 
   uint64_t getEntriesBufSize() const {
-    return *(uint64_t*)Buf;
+    uint64_t result;
+    std::memcpy(&result, Buf, sizeof result);
```

Comment:
This will always by uint64_t aligned at runtime, how about making this explicit by using reinterpret_cast ?

Reply:
Perhaps it's *supposed* to always be aligned, but in 21 tests, it isn't. [Here's the log, plus a full backtrace of one of the tests.](https://gist.github.com/brentdax/ea7be6748918e02815d2b8fd3be6e8e7)

Reply:
Ah thanks for clarifying! I misunderstood how ubsan for this works. LGTM.